src/widgets/RealtimeEffectPanel.cpp

```diff
@@ -0,0 +1,813 @@
+      dropHintLine->SetBackgroundColorIndex(clrDropHintHighlight);
+      dropHintLine->Hide();
+
+      rootSizer->Add(mEffectListContainer, 0, wxEXPAND);
+      rootSizer->Add(addEffect, 0, wxALIGN_CENTER | wxALL, 10);
+
+      SetSizer(rootSizer.release());
+
+      Bind(EVT_MOVABLE_CONTROL_DRAG_STARTED, [dropHintLine](const MovableControlEvent& event)
+      {
+         dropHintLine->Show();
+         dropHintLine->Raise();
+         //Make it zero-sized until first EVT_MOVABLE_CONTROL_DRAG_POSITION is triggered
+         dropHintLine->SetSize({0, 0});
+         if(auto window = dynamic_cast<wxWindow*>(event.GetEventObject()))
+            window->Raise();
+      });
+      Bind(EVT_MOVABLE_CONTROL_DRAG_POSITION, [this, dropHintLine](const MovableControlEvent& event)
+      {
+         constexpr auto DropHintLineHeihgh { 3 };//px
+
+         auto sizer = mEffectListContainer->GetSizer();
+         assert(sizer != nullptr);
+
+         if(event.GetSourceIndex() == event.GetTargetIndex())
+            //do not display hint line if position didn't change
+            dropHintLine->SetSize({0, 0});
+         else if(event.GetTargetIndex() == sizer->GetItemCount())
+         {
+            //insert at end
+            auto lastItem = sizer->GetItem(sizer->GetItemCount() - 1);
+            //avoid clipping
+            dropHintLine->SetPosition(lastItem->GetRect().GetBottomLeft() - wxPoint(0, DropHintLineHeihgh));
+            dropHintLine->SetSize(lastItem->GetSize().x, 3);
+         }
+         else if(auto item = sizer->GetItem(event.GetTargetIndex()))
+         {
+            dropHintLine->SetPosition(item->GetRect().GetTopLeft());
+            dropHintLine->SetSize(item->GetSize().x, 3);
+         }
+      });
+      Bind(EVT_MOVABLE_CONTROL_DRAG_FINISHED, [this, dropHintLine](const MovableControlEvent& event)
+      {
+         dropHintLine->Hide();
+         
+         auto trackOwner = mTrack->GetOwner();
+         if(!trackOwner)
+            return;
+
+         auto project = trackOwner->GetOwner();
+         if(project == nullptr)
+            return;
+
+         auto& effectList = RealtimeEffectList::Get(*mTrack);
+         auto from = event.GetSourceIndex();
+         auto to = event.GetSourceIndex() < event.GetTargetIndex() ? event.GetTargetIndex() - 1 : event.GetTargetIndex();
+         if(from != to)
+         {
+            effectList.MoveEffect(from, to);
+            ProjectHistory::Get(*project).PushState(
+               //i18n-hint: undo history, first parameter - track name
+               XO("'%s' effects reorder %d->%d").Format(mTrack->GetName(), from, to),
+               //i18n-hint: undo history record
+               XO("Realtime effect reorder"), UndoPush::CONSOLIDATE);
+         }
+         else
+         {
+            wxWindowUpdateLocker freeze(this);
+            Layout();
+         }
+      });
+      SetScrollRate(0, 20);
```

Comment:
This line isn't properly part of the commit for undo history so it might be put in an earlier commit